### ISSUE TITLE
Fix the build for docsy-as-submodule

### ISSUE THREAD

**Issue by chalin** [google/docsy#858]
_Originally posted by @LisaFC in https://github.com/google/docsy/discussions/535#discussioncomment-1963785_:

> Correct - feature is in but disabled by default, docs have not been updated yet.  Anyone who wants to try it can use the preview for the docs PR for more info. The docs update is sizeable and I want to take some time to do some organization work on it.

Based on the above comment, I believe that it wasn't @LisaFC's intention to merge #801 and introduce such a **serious breaking change**: #801 **broke the builds of most users**.

I believe that our first order of business is to (at least partially) **revert #801 so that all users can continue to work from HEAD**, as we continue to work through the details of how to introduce modules as an _optional_ feature (cf. #848).

We should also restore the build of the user-guide to be submodule based, since that reflects how most users are including docsy into their projects, and as such, it acts as a test that docsy@HEAD hasn't broken anything for submodule users.

@LisaFC @emckean @geriom @deining - thoughts.

**Comment by LisaFC** [comment]
Yes, it might be a good idea to either revert or (if it's nearly ready) #852 looks like it might be a way to fix things for everyone while still including the modules option (ie do what we thought #801 would do) - though it would need to have the submodule config as the default rather than one of two "equal options" to not break people's continuous deployment.

**Comment by chalin** [comment]
> #852 looks like it might be a way to fix things for everyone while still including the modules option (ie do what we thought #801 would do) - though it would need to have the submodule config as the default rather than one of two "equal options" to not break people's continuous deployment.

That works for me!

(We'll still need to refer the UG so that it builds from the local repo -- not a Hugo module.)

**Comment by deining** [comment]
> Based on the above comment, I believe that it wasn't @LisaFC's intention to merge #801 and introduce such a **serious breaking change**: #801 **broke the builds of most users**.

It wasn't my intention either, and I wasn't aware of the fact that it **would** break builds. Anyway, that's history now, let's look forwards.

I think finalizing #852 is the way to go, I just presented a [further idea](https://github.com/google/docsy/pull/852#issuecomment-1018728448) how to make it work in the desired way. Due to other private and job-related oblgaions, I couldn't work on this during the week as wanted. I will investigate the opportunities during the weekend and will post my findings shortly.

**Comment by chalin** [comment]
> It wasn't my intention either, and I wasn't aware of the fact that it **would** break builds.

Yes, of course, I believe you.

> I think finalizing #852 is the way to go, ... will present my findings shortly.

Great! Thanks for moving this forward.

**Comment by deining** [comment]
Another idea came to my mind: what about reverting #801, then adding a new directory `hugo-module`, recreating all folders from  root level (except for `userguide`) and then turn this subfolder `hugo-module` into a module. Inside this folder, I could create my own custom `config.toml`without getting in conflict with the `config.toml` at site root. I'm not sure this does work, but it could. Then users that want to use docsy as module have to write inside their site `config.toml`:

```
theme = ["github.com/google/docsy/hugo-module"]
```
This would break builds for users that migrated already. Not sure that we can bring #852 in a form that allows seamless update for migrated installs either.

WDYT of this? Would you prefer this over #852 in it's current form (incl. [amendments](https://github.com/google/docsy/pull/852#issuecomment-1018728448) as discussed).

**Comment by kneyugn** [comment]
> This would break builds for users that migrated already. Not sure that we can bring #852 in a form that allows seamless update for migrated installs either.

Hello! We have migrated, have users that will be affected. Keeping the current state or reverting to the original would be ideal.

**Comment by chalin** [comment]
> Another idea came to my mind: what about reverting #801,

Done via #860. As I mention there, apologies to all for the churn. We'll get this working, hopefully soon!

**Comment by deining** [comment]
> We'll get this working, hopefully soon!

We now have #871 in place. Within is PR docsy is used as submodule. [Preview](https://deploy-preview-871--docsydocs.netlify.app/) on Netlify is fine.

With this PR, the current name of the module was kept: `github.com/google/docsy`.
What needs to be done: you have to add in a second module/theme, which pulls in a workaround for a bug in Go's module management + the dependencies `bootstrap` and `Font-Awesome`.

```
theme = ["github.com/google/docsy", "github.com/google/docsy/module"]
```

**Comment by chalin** [comment]
Thanks @deining, I'll have a look at #871 shortly.

Closing this issue now. It was closed by #860.
